x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/user/SynonymsProfileUser.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+package org.elasticsearch.xpack.core.security.user;
+
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.xpack.core.security.authz.RoleDescriptor;
+import org.elasticsearch.xpack.core.security.support.MetadataUtils;
+
+import java.util.Map;
+
+/**
+ * internal user that manages the synonyms index.
+ */
+public class SynonymsProfileUser extends User {
```

Comment:
Nitpick: Would a better name for this be something like `SynonymsManagerUser` or `SynonymsAdminUser`?

Reply:
Changing to SynonymsInternalUser, to make it more clear it's an internal user - though there's not much of a standard here